File: app/src/main/java/com/asmat/rolando/bakingapp/db/AppDatabase.java
package com.asmat.rolando.bakingapp.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;


/**
 * Created by rolandoasmat on 8/7/17.
 */

@Database(entities = {ShoppingListIngredient.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    private static final String databaseName = "shopping_list_db";
    
    private static ShoppingListDao dbInstance;

    public abstract ShoppingListDao shoppingListDao();

    public static ShoppingListDao getInstance(Context context) {
        if(dbInstance == null) {
            dbInstance = Room.databaseBuilder(context, AppDatabase.class, databaseName).build().shoppingListDao();
        }
        return dbInstance;
    }
}

File: app/src/main/java/com/asmat/rolando/bakingapp/db/AppDatabase.kt
package com.asmat.rolando.bakingapp.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase


/**
 * Created by rolandoasmat on 7/16/17.
 */

@Database(entities = arrayOf(ShoppingListIngredient::class), version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun shoppingListDao(): ShoppingListDao

    companion object{
        private val databaseName = "shopping_list_db"
        var dbInstance: ShoppingListDao? = null

        fun getInstance(context: Context): ShoppingListDao? {
            if(dbInstance == null)
                // https://stackoverflow.com/questions/46665621/android-room-persistent-appdatabase-impl-does-not-exist
                dbInstance = Room.databaseBuilder(context, AppDatabase::class.java, databaseName).build().shoppingListDao()
            return dbInstance
        }
    }
}